client/src/pages/Main.js

```diff
@@ -1,32 +1,202 @@
+  Grid,
+  ListItemSecondaryAction,
+} from "@material-ui/core/";
 
-const useStyles = () => ({
-  RootGridContainer: {
+const useStyles = (theme) => ({
+  rightPart: {
     marginTop: "25px",
   },
   SwitchSelector: {
     height: "2.8em",
     width: "30%",
     fontWeight: 400,
   },
+  leftPanel: {
+    background: "white",
+    width: "100%",
+  },
+  platformItem: {
+    height: "6em",
+  },
+  platformFont: {
+    fontWeight: 800,
+  },
+  platformListItemAvatar: {
+    marginRight: "7px",
+  },
+  platformAvatar: {
+    width: theme.spacing(6),
+    height: theme.spacing(6),
+  },
+});
+
+const IOSSwitch = withStyles((theme) => ({
+  root: {
+    width: 36,
+    height: 19,
+    padding: 0,
+    margin: theme.spacing(1),
+  },
+  switchBase: {
+    padding: 1,
+    "&$checked": {
+      transform: "translateX(17px)",
+      color: theme.palette.common.white,
+      "& + $track": {
+        backgroundColor: "#6583f2",
+        opacity: 1,
+        border: "none",
+      },
+    },
+    "&$focusVisible $thumb": {
+      color: "#6583f2",
+      border: "6px solid #fff",
+    },
+  },
+  thumb: {
+    width: 17,
+    height: 17,
+  },
+  track: {
+    borderRadius: 26 / 2,
+    border: `1px solid ${theme.palette.grey[400]}`,
+    backgroundColor: "#bdc7d4",
+    opacity: 1,
+    transition: theme.transitions.create(["background-color", "border"]),
+  },
+  checked: {},
+  focusVisible: {},
+}))(({ classes, ...props }) => {
+  return (
+    <Switch
+      focusVisibleClassName={classes.focusVisible}
+      disableRipple
+      classes={{
+        root: classes.root,
+        switchBase: classes.switchBase,
+        thumb: classes.thumb,
+        track: classes.track,
+        checked: classes.checked,
+      }}
+      {...props}
+    />
+  );
 });
 
 class Main extends Component {
+  constructor(props) {
+    super(props);
+
+    // Get the params in the URL so we can know which platforms are selected.
+    let currentUrlParams = new URLSearchParams(window.location.search);
+    let selectedPlatformsInURL = currentUrlParams.get("platforms");
+    let splitSelectedPlatforms = [];
+    if (selectedPlatformsInURL) {
+      splitSelectedPlatforms = selectedPlatformsInURL.split(",");
+    }
+
+    this.state = {
+      allPlatforms: ["Reddit", "Twitter", "Facebook"],
+      platformSelected: [...splitSelectedPlatforms],
+    };
+  }
+
   render() {
     const { classes } = this.props;
 
+    const handlePlatformToggle = async (value) => {
+      // If a platform is already in "platformSelected", remove it.
+      // Otherwise, add it.
+      let index = this.state.platformSelected.indexOf(value.target.name);
+      if (index !== -1) {
+        let temp = this.state.platformSelected;
+        temp.splice(index, 1);
+        await this.setState({
+          platformSelected: temp,
+        });
+      } else {
+        await this.setState({
+          platformSelected: [...this.state.platformSelected, value.target.name],
+        });
+      }
```

Comment:
The toggles are off by default, so does that mean that there should be no mentions that are displayed when the user logs in?

Reply:
> The toggles are off by default, so does that mean that there should be no mentions that are displayed when the user logs in? just wondering if that might look strange if the right half is empty. Would it be enough to just add a message telling them to toggle on some feeds?

You are right,  I think a message will be helpful to teach users how to use. I will make this change soon.

Reply:
![image](https://user-images.githubusercontent.com/42278237/86544192-72733900-beea-11ea-95bf-516f35d31786.png)
Done. Please review it.

Reply:
That works! Thank you